tests/trx_generator/trx_generator.cpp

```diff
@@ -166,10 +168,102 @@ namespace eosio::testing {
       }
    }
 
-   trx_generator::trx_generator(std::string chain_id_in, const std::string& abi_data_file, std::string contract_owner_account, std::string auth_account, std::string action_name,
-         const std::string& action_data_file_or_str, fc::microseconds trx_expr, const std::string& private_key_str, std::string lib_id_str, std::string log_dir, bool stop_on_trx_failed, const std::string& peer_endpoint, unsigned short port)
-       : trx_generator_base(chain_id_in, contract_owner_account, trx_expr, lib_id_str, log_dir, stop_on_trx_failed, peer_endpoint, port), _abi_data_file_path(abi_data_file), _auth_account(auth_account),
-         _action(action_name), _action_data_file_or_str(action_data_file_or_str), _private_key(fc::crypto::private_key(private_key_str)) {}
+   void trx_generator::locate_key_words_in_action_mvo(std::vector<std::string>& acct_gen_fields_out, fc::mutable_variant_object& action_mvo, const std::string& key_word) {
+      for(const mutable_variant_object::entry& e: action_mvo) {
+         if(e.value().get_type() == fc::variant::string_type && e.value() == key_word) {
+            acct_gen_fields_out.push_back(e.key());
+         } else if(e.value().get_type() == fc::variant::object_type) {
+            auto inner_mvo = fc::mutable_variant_object(e.value());
+            locate_key_words_in_action_mvo(acct_gen_fields_out, inner_mvo, key_word);
+         }
+      }
+   }
+
+   void trx_generator::locate_key_words_in_action_array(std::map<int, std::vector<std::string>>& acct_gen_fields_out, fc::variants& action_array, const std::string& key_word) {
+      for(size_t i = 0; i < action_array.size(); ++i) {
+         auto action_mvo = fc::mutable_variant_object(action_array[i]);
+         locate_key_words_in_action_mvo(acct_gen_fields_out[i], action_mvo, key_word);
+      }
+   }
+
+   void trx_generator::update_key_word_fields_in_sub_action(std::string key, fc::mutable_variant_object& action_mvo, std::string action_inner_key, const std::string key_word) {
+      auto mvo = action_mvo.find(action_inner_key);
+      if(mvo != action_mvo.end()) {
+         fc::mutable_variant_object inner_mvo = fc::mutable_variant_object(action_mvo[action_inner_key].get_object());
+         if (inner_mvo.find(key) != inner_mvo.end()) {
+            inner_mvo.set(key, key_word);
```

Comment:
`std::move(key)` Although, I think just changing the function to `const std::string&` would be better.

Reply:
Addressed: 1ad088899785
used `const std::string&`